### PR TITLE
Add tests.yaml for github actions

### LINKED ISSUE #266: Switch from Travis CI to Github Actions
I want to investigate switching our CI to Github Actions. Using Travis CI is very frustrating because these yaml files and their behavior on Travis is really hard to figure out. Also, Open-Source projects have become deprioritized on Travis, see e.g. this link: https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works

### PR DESCRIPTION
**Pull request by mschmidt87**
This PR ports our Continuous Integration workflow from Travis CI to github actions:

- Adds two files, one for the test suite, one for the publication of the documentation (only on pull-request merge)
- Deletes the .travis configuration file
- Deletes the `deploy.sh` script because I integrated this into the workflows file
- Adds a config file for coveralls required by the github action managing coveralls.

The functionality of the CI itself does not change with this PR.

Note: Currently, we still require travis to pass in order to merge a PR. Once this PR is approved, I will deactivate this setting and then merge the PR. After that, I'll activate checks that the new workflow passes for all PRs.

Closes #266

### CODE REVIEW

File: .github/workflows/deploy_docs.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python3 -m pip install --upgrade pip
```
Comment (jakobj) [inline .github/workflows/deploy_docs.yaml:19]: why does this need `python3`? should just `python` work in this env? would keep this more general

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        env:
+          DEP: ${{ matrix.dep }}
+        run: |
+          python3 -m pip install --upgrade pip
```
Comment (jakobj) [inline .github/workflows/tests.yaml:33]: see above

**Review by jakobj — APPROVED** [review]
looks awesome, thanks for taking care of this! :rocket: 

one small question. and just one thing confuses me a bit: where do you configure what is run when? is this just set on the github interface?

**Comment by mschmidt87** [comment]
> one small question. and just one thing confuses me a bit: where do you configure what is run when? is this just set on the github interface?

I think by default github runs all workflows defined in `.github/workflows/`. Each of these workflows defines when it's run, in the `on` field of the yaml file.

**Pull request merged**